src/NServiceBus.Core/Forwarding/ForwardingToDispatchConnector.cs

```diff
@@ -0,0 +1,16 @@
+namespace NServiceBus
+{
+    using System;
+    using NServiceBus.Forwarding;
+    using NServiceBus.Pipeline;
+    using NServiceBus.TransportDispatch;
+    using NServiceBus.Transports;
+
+    class ForwardingToDispatchConnector:StageConnector<ForwardingContext,DispatchContext>
+    {
+        public override void Invoke(ForwardingContext context, Action<DispatchContext> next)
+        {
+            next(new DispatchContext(context.Get<OutgoingMessage>(),context));
```

Comment:
How about a vocabulary like Fork and Join and appropriate classes. So connector becomes `BlaJoin` and creating a new pipeline becomes `BlaFork`? Having multiple `PipelineBase` makes the code really complex. Of course we should also do a rename of our current concepts. Intention reveiling names